api/base/exceptions.py

```diff
@@ -34,6 +34,13 @@ def json_api_exception_handler(exc, context):
 
         response.data = {'errors': errors}
 
+    # Return 401 instead of 403 during unauthorized requests without having user log in with Basic Auth
+    error_message = response.data['errors'][0].get('detail')
+    errors_401 = ["Authentication credentials were not provided.", 'Incorrect authentication credentials.']
```

Comment:
This appears to be more of a workaround then the real fix to the problem.

`NotAuthenticated` should be raised where appropriate; doing so will result in the 401 responses that we want.